specs/unified_kernel_image.md

```diff
@@ -159,3 +159,41 @@ that tools staging or consuming UKIs have a common place to store and look for t
 
 The installed UKIs should have a filename `<version format specification>.efi`, i.e. the filename is left to
 implementers but must be valid for comparisons according to the [Version Format Specification](version_format_specification.md).
+
+### Auxiliary resources
+
+Auxiliary UKI resources (such as PE addons for command line extensions and similar,
+as well as systemd-sysext and systemd-confext DDIs) built centrally by distributions and
+installed via package manager should be installed into locations depending on
+whether they should be applied to all UKIs installed in the ESP, or only to a
+single specific UKI.
+
+UKI auxiliary resources that shall apply to *all* installed kernels may be
+installed into `/usr/lib/modules/uki.extra.d/`. UKI auxiliary resources that
+shall apply to *one* specific installed kernel shall be instead installed into
+`/usr/lib/modules/$UNAME/$UKI.efi.extra.d/`, where `$UNAME` is the output of
+`uname -r` of the kernel included in the UKI and `$UKI` is the name of the
+corresponding centrally built UKI with the `.efi` extension stripped.
+
+The installed UKI auxiliary resources must have a specific file extension, which
+depends on the resource type:
+* for PE addons is `*.addon.efi`
+* for sysext DDIs is `*.sysext.raw`
+* for confext DDIs is `*.confext.raw`
+
+
+### Example
+Given the following parameters:
+* `ARCH=x86_64`
+* `DISTRO=fedora`
+* `UKI="bar.efi"`
+* `UKI_KERNEL_VERSION="6.9.1-1-foo"`
+* `RESOURCE_NAME="test"`
+* The resource type is `PE addon`, meaning the file extension is `.addon.efi`
```

Comment:
Since the extension follows from the type, the part after the comma would be better in the second paragraph, but I'd just drop it, since it appears there already and it follows from the text and the parameters given.